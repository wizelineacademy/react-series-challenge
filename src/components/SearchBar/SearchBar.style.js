import styled from 'styled-components'

export const StyledInput = styled.input`
  width: 920px;
  height: 20px;
  font-size: 15px;
  padding: 15px;
  margin-bottom: 14px;
  font-family: sans-serif;
  font-weight: 400;

  :focus {
    outline: none;
  }
`
export const StyledForm = styled.form`
  display: flex;
  justify-content: center;
`
StyledForm.displayName = 'form'
