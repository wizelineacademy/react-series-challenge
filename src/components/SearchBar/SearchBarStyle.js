import styled from "styled-components";

const FormStyle = styled.form`
    display: flex
`;

const ButtonSearchStyle = styled.button`
    border: none;
    background: linear-gradient(-90deg, #8E43FD, #E086D3);
    border-radius: 0 1rem 1rem 0;
    color: white;
    font-size: 2rem;
    padding: 0 1rem;
`;

const SearchBarStyle = styled.input`
    font-size: 2rem;
    width: 200px;
    border-radius: 1rem 0 0 1rem;
    padding: 0.5rem 3rem;
    border: none;
    transition: ease-in-out, width .35s ease-in-out;

    &:focus {
        width: 250px;
        outline: none
    }
`;


export { SearchBarStyle, ButtonSearchStyle, FormStyle };