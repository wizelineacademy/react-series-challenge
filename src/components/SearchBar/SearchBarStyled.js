import styled from 'styled-components';

const Input = styled.input`
  padding: 0.5em;
  margin: 0.5em;
  color: darkblue;
  background: white;
  border: none;
  border-radius: 8px;
  min-width: 350px;
  min-height: 30px;
  text-align: center;
  font-size: 1em;
  font-weight: bold;
`;

export default Input;
