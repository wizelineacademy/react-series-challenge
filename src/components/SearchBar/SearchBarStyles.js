import styled from 'styled-components';

const SearchBarContainer = styled.div`
  display: inline-block;
  order: 2;
  flex: 0 1 auto;
  align-self: auto;
  min-width: 0;
  min-height: auto;
`;

const SearchBarBox = styled.input`
  width: 375px;
  display: block;
  border: 1px solid #999;
  height: 25px;
  -webkit-box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.3);
  -moz-box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.3);
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.3);
`;

export { SearchBarContainer, SearchBarBox };
