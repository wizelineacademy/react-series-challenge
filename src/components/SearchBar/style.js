import styled from 'styled-components'

export const Container = styled.div `
  display: flex;
  justify-content: center;
  margin-bottom: 70px;
`

export const SearchBox = styled.div `
  position: absolute;
  input {
    border: none;
    background: none;
    z-index: 1;
    width: 50px;
    height: 50px;
    transition: all .25s ease-in .25s;
    color: transparent;
    font-size: 1.25rem;
    line-height:50px;
    &:hover {
      cursor: pointer;
      &:focus {
        cursor: text;
      }
      + span {
        background: rgba(255,255,255,0.2);
      }
    }
    &:focus {
      width: 200px;
      padding: 0 10px;
      outline: none;
      color: black;
      background:none;
      color: white;
      + span {
        width: 200px;
        &::before {
          width:2px;
          opacity: 0;
          transition: all .25s ease-in;
        }
      }
    }
    + span {
      z-index: -1;
      position: absolute;
      border: 2px solid white;
      top: 0;
      width: 50px;
      height: 50px;
      transition: all .25s ease-in .25s;
      border-radius: 50px;
      left: 0;
      &::before {
        transition: all .25s ease-in .5s;
        transform-origin: left top;
        content: '';
        position: absolute;
        width: 20px;
        height: 7px;
        border-radius: 5px;
        background: white;
        transform: rotate(45deg) translate(43px, 10px);
      }
    }
  }
`