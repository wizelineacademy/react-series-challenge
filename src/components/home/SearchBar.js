/**
 * SearchBar Component
 */

import React, { Component } from 'react';

/**
 * SearchBar Component becomes a class based component to handle its value state
 */

export default class SearchBar extends Component {
    constructor(props){
        super(props);

        this.state = {
            term: ""
        }
    }

    onTermChange = (e) => {
        const targetElement = e.target;
        const newValue = targetElement.value;
        
        this.setState( () => {
            return { term: newValue }
        } );
    }

    render(){
        return (
            <section className="SearchBar">
                <input type="text" value={this.state.term} onChange={ this.onTermChange } placeholder="Search Giphy:" />
            </section>
        );
    };
}