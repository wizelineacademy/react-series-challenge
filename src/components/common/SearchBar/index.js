import SearchBar from './StyledSearchBar';

export default SearchBar
