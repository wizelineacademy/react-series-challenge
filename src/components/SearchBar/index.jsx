import React from 'react'
import { func, string } from 'prop-types'
import Icon from 'react-icons-kit'
import { search } from 'react-icons-kit/icomoon/search'

import { StyledInput, StyledForm } from './SearchBar.style'

const SearchBar = ({ onChange, onSubmit, value }) => (
  <div>
    <StyledForm onSubmit={onSubmit}>
      <StyledInput onChange={onChange} value={value} type='text' name='search' />
      <button type='submit' ><Icon icon={search} /></button>
    </StyledForm>
  </div>
)

SearchBar.propTypes = {
  onChange: func.isRequired,
  onSubmit: func.isRequired,
  value: string
}
SearchBar.defaultProps = {
  value: ''
}
export default SearchBar
