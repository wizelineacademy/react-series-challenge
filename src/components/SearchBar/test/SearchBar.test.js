import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchBar from '..';
const props = {
  isFavorite: true,
  gif: {
    title:'title',
    images: {
      downsized_large: {
        url: 'test'
      }
    }
  },
  searchGifs: jest.fn(),
  setFavorite: jest.fn(),
  history: {
    push: jest.fn()
  },
  location: {
    pathname: '/'
  },
  match: {
    params: {
      id: 1
    }
  }
}

describe('SearchBar', () => {
  it('should render correctly', () => {
    const output = shallow(
      <SearchBar.WrappedComponent {...props} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should be defined', () => {
   expect(SearchBar).toBeDefined();
  });
  it('should submit when no favorite', () => {
    const output = shallow(
      <SearchBar.WrappedComponent {...props} />
    );
    output.find('form').simulate('submit', {
      preventDefault: () => {}
    })
    expect(props.searchGifs).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalled();
  })
  it('should submit when favorite', () => {
    const favoriteLocation = {
      pathname: '/favorites'
    }
    const output = shallow(
      <SearchBar.WrappedComponent {...props} location={favoriteLocation} />
    );
    output.find('form').simulate('submit', {
      preventDefault: () => {}
    })
    expect(props.searchGifs).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalled();
  })
});
