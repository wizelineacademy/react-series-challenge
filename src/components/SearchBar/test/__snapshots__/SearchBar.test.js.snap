// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <styled.input
    onChange={[Function]}
    placeholder="Search all the GIFs"
    value=""
  />
</form>
`;
